### DISCUSSION TITLE
My chatflows are deleting / not being found after a while. Paying in Render to deploy. Please help!

### DISCUSSION THREAD

**Discussion by camilordg96** [FlowiseAI/Flowise#232] [Q&A] [answered]
Hi all, I deployed FLowise through render.com and am paying the $7 dollar starter. I create the chatflows and after a couple of minutes (30 let's say) the flows are deleted. Has this happened to anyone, any advice?

**Comment by camilordg96** [comment]
<img width="588" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/98601378/52fe772f-689e-4aed-b55f-3d62d0012dbb">
 this is the error I'm getting: Failed to retrieve chatflow: Chatflow c427a9fb-edc3-4d99-a39f-241c780f3e88 not found

**Comment by og56802409819745** [comment]
i have the same issue, if i will manage to fix it, i will text it here

**Comment by og56802409819745** [comment]
Founded sollution on flowise discord: Railway and Render are ephemeral, everything is wiped after redeploy, so remember to export your database first before redeploy!

**Reply by marius-smg** [reply]
Hey, maybe a stupid question, but what do you mean by export database before redeploy? As i understand, i dont know at which time render resets, so redeployment is unintentional. What exactly do you mean by database? the Database (in my case on pinecone) is not affacted by that. It's just the chatflows that get wiped after render resets.

**Reply by AutoAlchemists** [reply]
Did you manage to resolve? have the same exact issue

**Reply by Kvkthecreator** [reply]
yea, you have to update git commit to latest one and re-connect all your workflows unfortunately. It seems it was a bug resolved with the latest update

**Comment by tejaswiniallikanti** [comment]
But does it happen when I do it in local?

**Reply by janjarco** [reply]
I have similar issue working with Azure. After about 15 minutes I cannot access my chat flows, they are just deleted. It works well on local machine

**Reply by Kvkthecreator** [reply]
i have exact same issue. (both on Azure, and on github codespaces) :(
hopefully someone can find solution, as a temp workaround i'm saving my chatflows / database as backups.
it sucks cuz i have to reconnect my APIs everytime.

**Comment by tejaswiniallikanti** [comment]
Yes, it works fine in local. The vector database doesnt change, but the
chatflows get deleted. I want to you use the chatflows APIs in other
application and I am not able to do it.

On Wed, Jun 21, 2023 at 1:36 AM Jan Jarco ***@***.***> wrote:

> I have similar issue working with Azure. After about 15 minutes I cannot
> access my chat flows, they are just deleted. It works well on local machine
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/FlowiseAI/Flowise/discussions/232#discussioncomment-6234771>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADMOZM2QSMGLKGZ7ZHIG4YLXMH7EJANCNFSM6AAAAAAYUWOJPY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Regards,
Tejaswini Allikanti

**Comment by tejaswiniallikanti** [comment]
So presently, how are you deploying it? You are updating the APIs all the
time?

On Wed, Jun 21, 2023 at 11:33 AM Kvkthecreator ***@***.***>
wrote:

> i have exact same issue. (both on Azure, and on github codespaces) :(
> hopefully someone can find solution, as a temp workaround i'm saving my
> chatflows / database as backups.
> it sucks cuz i have to reconnect my APIs everytime.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/FlowiseAI/Flowise/discussions/232#discussioncomment-6237943>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADMOZM2UX3AOB47MM72BKF3XMKFEVANCNFSM6AAAAAAYUWOJPY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Regards,
Tejaswini Allikanti

**Comment by HenryHengZJ** [comment, marked as answer]
In the latest v1.2.13 release, you can specify an env DATABASE_PATH. This will allow you to specify where the data is being stored. And using Render's Disk, you can mount that path and have persistent storage.

Here's the step by step guide - https://docs.flowiseai.com/deployment/render#persistent-disk

Let us know if that resolves the issue
